### ISSUE TITLE
Schema issue with new Joi

### ISSUE THREAD

**Issue by xenolalia** [sebelga/gstore-node#175]
Following code was working fine when I was using old 'joi' package. Then I installed new '@hapi/joi' and replaced the references, it still worked. But then I uninstalled old 'joi' package, and it started failing. Am I missing something or is it a bug?

```
const Joi = require('@hapi/joi');
const { gstore } = require('../db');
const { Schema } = gstore;

const userSchema = new Schema({
    username: { joi: Joi.string().required() },
}, { joi: true });
```

Cannot read property 'object' of null : TypeError: Cannot read property 'object' of null
    at buildJoiSchema (..\\node_modules\\gstore-node\\lib\\schema.js:243:25)
    at new Schema (..\\node_modules\\gstore-node\\lib\\schema.js:94:25)
    at Object.<anonymous> (..\\src\\users\\user.model.js:28:23)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (..\\src\\users\\users.handler.js:2:27)

**Comment by sebelga** [comment]
Hi,

Thanks for raising the issue. Could you confirm that you have indeed the folder `@hapi/joi` in your node_modules. It seems that Joi was not found from the error you provided.

Cheers! 👍

**Comment by xenolalia** [comment]
Hi, Thanks for responding!

1. I removed entire node_modules folder
2. Re-installed all packages using npm.
3. @hapi/joi is installed. And I verified in the node_modules folder.
4. My project only uses '@hapi/joi'. I do not have any references to the old 'joi' package.

And I'm still getting the exact same error.

As soon as I install old 'joi' package along with new '@hapi/joi', the error goes away. Is there any chance that gstore-node is still looking for old 'joi' package?

**Comment by xenolalia** [comment]
I've "gstore-node": "^6.0.2" in my package.json, which seems to be the latest version

I checked under my project's node_modules -> gstore-node -> package.json, where it shows devDependencies as "joi": "^12.0.0"

But then I checked your github page -> package.json, where it shows devDependencies as "@hapi/joi": "^15.1.1"

Could this be an issue?

**Comment by xenolalia** [comment]
Seems like I may have found the issue. The latest release is 6.0.2 which was published on 2019-04-26. But you have a commit "chore: added support for @hapi/joi package" on 2019-05-22.

That means the current version of gstore-node is still relying on the deprecated version of joi. I guess this problem will get resolved once the new version comes with the new @hapi/joi package.

Do you have any plans for the next release?

**Comment by sebelga** [comment]
Thanks for looking into this. I indeed _thought_ it had been released but the commit came after `6.0.2`. I am going to make a release now, it has a few breaking changes so it will be `v7.0.0`.

I will close this issue, feel free to re-open it if the problem persists. Cheers! 👍
